test/linalg/test_linalg_lstsq.fypp

```diff
@@ -100,6 +102,39 @@ module test_linalg_least_squares
     
     #:endif
     #:endfor
+    
+    ! Test issue #823
+    subroutine test_issue_823(error)
+        type(error_type), allocatable, intent(out) :: error
+        
+        ! Dimension of the problem.
+        integer(ilp), parameter :: n = 42
+        ! Data for the least-squares problem.
+        complex(dp) :: A(n+1, n), b(n+1), x_true(n), x_lstsq(n+1)
```

Comment:
x_lstsq should be of size `x_lstsq(n)`. Tested it locally with the proposed fix and it works ... had to use n+1 in Compiler Explorer as it was not working there

Reply:
Thanks @jalvesz: [89dc992](https://github.com/fortran-lang/stdlib/pull/830/commits/89dc99273cb3a86e901362a72154c39288a6eeda)